src/scales/scale.category.js

```diff
@@ -49,61 +62,58 @@ module.exports = Scale.extend({
 	},
 
 	// Used to get data value locations.  Value can either be an index or a numerical value
-	getPixelForValue: function(value, index) {
+	getPixelForValue: function(value, index, datasetIndex) {
 		var me = this;
 		var offset = me.options.offset;
+
 		// 1 is added because we need the length but we have the indexes
-		var offsetAmt = Math.max((me.maxIndex + 1 - me.minIndex - (offset ? 0 : 1)), 1);
+		var offsetAmt = Math.max(me.maxIndex + 1 - me.minIndex - (offset ? 0 : 1), 1);
+
+		var isHorizontal = me.isHorizontal();
+		var valueDimension = (isHorizontal ? me.width : me.height) / offsetAmt;
```

Comment:
I noticed `isHorizontal ? me.width : me.height` being done a couple times in the most recent PR as well. Perhaps we should add a `_getLength()` method? That could be a separate PR though since it would probably simplify all scales

Reply:
I was also thinking about `_getRange()` that returns `{start, end, size}` to eliminate `isHorizontal ? me.left : me.top` as well. I agree that it should be done in a separate PR.